test/e2e/runner/load.go

```diff
@@ -66,28 +64,41 @@ func Load(ctx context.Context, testnet *e2e.Testnet, multiplier int) error {
 	}
 }
 
-// loadGenerate generates jobs until the context is canceled
-func loadGenerate(ctx context.Context, chTx chan<- types.Tx, multiplier int, size int64) {
-	for i := 0; i < math.MaxInt64; i++ {
-		// We keep generating the same 100 keys over and over, with different values.
-		// This gives a reasonable load without putting too much data in the app.
-		id := i % 100
-
-		bz := make([]byte, size)
-		_, err := rand.Read(bz)
-		if err != nil {
-			panic(fmt.Sprintf("Failed to read random bytes: %v", err))
-		}
-		tx := types.Tx(fmt.Sprintf("load-%X=%x", id, bz))
+// loadGenerate generates jobs until the context is canceled.
+//
+// The chTx has multiple consumers, thus the rate limiting of the load
+// generation is primarily the result of backpressure from the
+// broadcast transaction, though at most one transaction will be
+// produced every 10ms.
+func loadGenerate(ctx context.Context, chTx chan<- types.Tx, size int64) {
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+	defer close(chTx)
 
+	for {
 		select {
-		case chTx <- tx:
-			sqrtSize := int(math.Sqrt(float64(size)))
-			time.Sleep(10 * time.Millisecond * time.Duration(sqrtSize/multiplier))
-
 		case <-ctx.Done():
-			close(chTx)
 			return
+		case <-timer.C:
```

Comment:
nit: we could avoid nesting the select below in this case since the other case here returns. Falling through to the rest of the logic in this loop when the timer case is hit works just as well be avoids the nesting.

Reply:
what if the context gets canceled between when the timer fires and before the chTx channel has capacity for the transaction we produce?

Reply:
Ah, sorry, to clarify: We could just un-nest all of this code by having this case simply fall through. I.e.:

```go
select {
		case <-ctx.Done():
			return
		case <-timer.C:
}

// create  the tx here

select {
case <-ctx.Done():
	return
case chTx <- tx:
	// sleep for a bit before sending the
	// next transaction.
	waitTime := (10 * time.Millisecond) + time.Duration(rand.Int63n(int64(250*time.Millisecond))) // nolint: gosec
	timer.Reset(waitTime)
}
```
This is mostly a style change to avoid nesting code.